src/test/java/io/personium/test/jersey/box/odatacol/UserDataListFilterTypeValidateTest.java

```diff
@@ -1030,6 +1030,81 @@ public UserDataListFilterTypeValidateTest() {
+
+        query = "?\\$filter=datetime+lt+datetime'2015-01-07T00:19:16.173'&\\$inlinecount=allpages";
+        res = UserDataUtils.list(CELL, BOX, COL, ENTITY, query, TOKEN, HttpStatus.SC_OK);
+        // should match 1
+        ODataCommon.checkResponseBodyList(res.bodyAsJson(), null, NAMESPACE, null, 1);
+
+        query = "?\\$filter=datetime+le+datetime'2015-01-07T00:19:16.173'&\\$inlinecount=allpages";
+        res = UserDataUtils.list(CELL, BOX, COL, ENTITY, query, TOKEN, HttpStatus.SC_OK);
+        // should match 2
+        ODataCommon.checkResponseBodyList(res.bodyAsJson(), null, NAMESPACE, null, 2);
+
+
+        query = "?\\$filter=datetime+gt+datetime'2015-01-07T00:19:16.173'&\\$inlinecount=allpages";
+        res = UserDataUtils.list(CELL, BOX, COL, ENTITY, query, TOKEN, HttpStatus.SC_OK);
+        // should match 0
+        ODataCommon.checkResponseBodyList(res.bodyAsJson(), null, NAMESPACE, null, 0);
+
+        query = "?\\$filter=datetime+ge+datetime'2015-01-07T00:19:16.173'&\\$inlinecount=allpages";
+        res = UserDataUtils.list(CELL, BOX, COL, ENTITY, query, TOKEN, HttpStatus.SC_OK);
+        // should match 1
+        ODataCommon.checkResponseBodyList(res.bodyAsJson(), null, NAMESPACE, null, 1);
+
+
+        query = "?\\$filter=datetime+gt+datetime'2015-01-07T00:19:16.172'&\\$inlinecount=allpages";
+        res = UserDataUtils.list(CELL, BOX, COL, ENTITY, query, TOKEN, HttpStatus.SC_OK);
+        // should match 1
+        ODataCommon.checkResponseBodyList(res.bodyAsJson(), null, NAMESPACE, null, 1);
+
+        query = "?\\$filter=datetime+ge+datetime'2015-01-07T00:19:16.172'&\\$inlinecount=allpages";
+        res = UserDataUtils.list(CELL, BOX, COL, ENTITY, query, TOKEN, HttpStatus.SC_OK);
+        // should match 2
+        ODataCommon.checkResponseBodyList(res.bodyAsJson(), null, NAMESPACE, null, 2);
+
+
+
+        // should match 0
+        query = "?\\$filter=datetime+lt+datetime'1970-01-01T00:00:00.900'&\\$inlinecount=allpages";
+        res = UserDataUtils.list(CELL, BOX, COL, ENTITY, query, TOKEN, HttpStatus.SC_OK);
+        ODataCommon.checkResponseBodyList(res.bodyAsJson(), null, NAMESPACE, null, 0);
+
+        // should match 0
+        query = "?\\$filter=datetime+lt+datetime'2000-01-01T00:00:00.000'&\\$inlinecount=allpages";
+        res = UserDataUtils.list(CELL, BOX, COL, ENTITY, query, TOKEN, HttpStatus.SC_OK);
+        ODataCommon.checkResponseBodyList(res.bodyAsJson(), null, NAMESPACE, null, 0);
+
+        // should match 0
+        query = "?\\$filter=datetime+gt+datetime'2020-01-01T00:00:00.000'&\\$inlinecount=allpages";
+        res = UserDataUtils.list(CELL, BOX, COL, ENTITY, query, TOKEN, HttpStatus.SC_OK);
+        ODataCommon.checkResponseBodyList(res.bodyAsJson(), null, NAMESPACE, null, 0);
+
+        // should match 0
+        query = "?\\$filter=__published+lt+datetime'1970-01-01T00:00:00.900'&\\$inlinecount=allpages";
```

Comment:
This is actually intentional. 
## Why __published instead of datetime? 
-  __published is another DateTime type property. 
-  Since this property is system defined, as opposed to user defined properties such as "datetime" in this case,  it is treated differently in personium-core
-  So we had better have tests for it.  
## Why not 000 but 900 ?
- "1970-01-01T00:00:00.900" is interpreted as unix time 900. 
-  "900" is greater than "1420589956172" if compared as String whereas "0" is not. 
- This test is intended to make sure the personium-core uses "number" index, not "text" index.

Reply:
Probably it is better to write these in the test case comment. Thanks.

Reply:
Updated comments